### PR TITLE
feat(build): bump dependencies for the given branch

### PR DESCRIPTION
**Pull request by link108**
Related to https://github.com/spinnaker/bumpdeps/pull/17
Pass the branch to bump dependencies so we can bump dependencies for a given branch

### CODE REVIEW

File: .github/workflows/bump_dependencies.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ jobs:
     - uses: spinnaker/bumpdeps@master
       with:
         ref: ${{ github.event.client_payload.ref }}
+        branch: ${{ github.event.client_payload.branch }}
```
Comment (dbyron-sf) [inline .github/workflows/bump_dependencies.yml:14]:
```suggestion
        baseBranch: ${{ github.event.client_payload.branch }}
```
?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -15,6 +15,11 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - run: git fetch --prune --unshallow
+      - name: Get Branch
```
Comment (dbyron-sf) [inline .github/workflows/release.yml:18]: Perhaps a comment about why we're doing this / what it's doing...and perhaps some mention of what happens when there's more than one or no branches?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -15,6 +15,16 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - run: git fetch --prune --unshallow
+      # Given a tag, determine what branch we are on
+      # Minor tags (eg v0.1.0) should be pushed to the master branch
```
Comment (dbyron-sf) [inline .github/workflows/release.yml:19]: I'm not sure this distinction between minor and patch is helpful here.  I mean, the code here doesn't have any logic to figure out minor/patch/etc.  It's figuring out the current branch (e.g. master or release-x) and targeting the autobump PR to that branch, assuming the branch naming convention is the same in the consumers of kork as it is in kork.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -15,6 +15,16 @@ jobs:
...
+      # Patch tags (eg v0.1.1+) should be pushed to their respective release branches
+      # If the tag doesn't exist on a branch, bumpDeps will log and throw an error
+      # TODO: handle error case if tag exists on multiple branches
```
Comment (dbyron-sf) [inline .github/workflows/release.yml:22]: It'd be nice to handle the no-branch case here too, and potentially abort and not create the github release nor publish a jar in this case.

**Review by dbyron-sf — APPROVED** [review]

**Comment by link108** [comment]
@mergifyio backport release-1.24.x

**Pull request merged**